net/dns/resolver/forwarder.go

```diff
@@ -601,17 +597,6 @@ type forwardQuery struct {
 	// ...
 }
 
-// forward forwards the query to all upstream nameservers and waits for
-// the first response.
-//
-// It either sends to f.responses and returns nil, or returns a
-// non-nil error (without sending to the channel).
-func (f *forwarder) forward(query packet) error {
-	ctx, cancel := context.WithTimeout(f.ctx, responseTimeout)
```

Comment:
@twitchyliquid64 `responseTimeout` is now unused/unenforced, was this an intentional change?

(noticed while discussing #5308 with @fylie, it seemed like that 5 second timeout might explain observed behavior, but we're not actually enforcing it anywhere).

Reply:
The timeout was refactored into `Resolver`: See tsdns.go:281. The other timeout is on the TCP DNS path (45 seconds for an idle connection), in `manager.go` ~line 415.

Reply:
Thanks, that makes sense. Sent out #5629 to clean it up.